arex-instrumentation/httpclient/arex-httpclient-ning/src/main/java/io/arex/inst/httpclient/ning/AsyncHttpClientInstrumentation.java

```diff
@@ -0,0 +1,81 @@
+
+import com.ning.http.client.ListenableFuture;
+import com.ning.http.client.Request;
+import com.ning.http.client.Response;
+import io.arex.agent.bootstrap.model.MockResult;
+import io.arex.inst.extension.MethodInstrumentation;
+import io.arex.inst.extension.TypeInstrumentation;
+import io.arex.inst.httpclient.common.HttpClientAdapter;
+import io.arex.inst.runtime.context.ContextManager;
+import io.arex.inst.runtime.context.RepeatedCollectManager;
+import io.arex.inst.runtime.listener.DirectExecutor;
+import io.arex.inst.runtime.util.IgnoreUtils;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.implementation.bytecode.assign.Assigner;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import java.util.Collections;
+import java.util.List;
+
+import static net.bytebuddy.matcher.ElementMatchers.*;
+
+public class AsyncHttpClientInstrumentation extends TypeInstrumentation {
+    @Override
+    protected ElementMatcher<TypeDescription> typeMatcher() {
+        return named("com.ning.http.client.AsyncHttpClient");
+    }
+
+    @Override
+    public List<MethodInstrumentation> methodAdvices() {
+        return Collections.singletonList(new MethodInstrumentation(
+                isMethod().and(named("executeRequest").and(takesArguments(2))), ExecuteRequestAdvice.class.getName()));
+    }
+
+    public static class ExecuteRequestAdvice {
+        @Advice.OnMethodEnter(suppress = Throwable.class, skipOn = Advice.OnNonDefaultValue.class)
+        public static boolean onEnter(@Advice.Argument(0) Request request,
+                                      @Advice.Local("mockResult") MockResult mockResult,
+                                      @Advice.Local("extractor") NingHttpClientExtractor extractor){
+            if (IgnoreUtils.excludeOperation(request.getUri().getPath())) {
+                return false;
+            }
+            if (ContextManager.needRecord()) {
+                RepeatedCollectManager.enter();
+            }
+            if (ContextManager.needRecordOrReplay()) {
+                HttpClientAdapter<Request, Response> adapter = new NingHttpClientAdapter(request);
+                extractor = new NingHttpClientExtractor(adapter);
+                if (ContextManager.needReplay()) {
+                    mockResult = extractor.replay();
+                    return mockResult != null && mockResult.notIgnoreMockResult();
+                }
+            }
+            return false;
+        }
+
+        @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class)
+        public static void onExit(@Advice.Enter boolean skipOnEnter,
+                                  @Advice.Local("mockResult") MockResult mockResult,
+                                  @Advice.Local("extractor") NingHttpClientExtractor extractor,
+                                  @Advice.Return(readOnly = false, typing = Assigner.Typing.DYNAMIC) ListenableFuture responseFuture,
```

Comment:
should not need `typing = Assigner.Typing.DYNAMIC`, only uncertain types, such as Object class